platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/state/service/WritableRosterStoreTest.java

```diff
@@ -0,0 +1,162 @@
+import static org.mockito.Mockito.when;
+
+import com.hedera.hapi.node.base.ServiceEndpoint;
+import com.hedera.hapi.node.state.primitives.ProtoBytes;
+import com.hedera.hapi.node.state.roster.Roster;
+import com.hedera.hapi.node.state.roster.RosterEntry;
+import com.hedera.hapi.node.state.roster.RosterState;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import com.swirlds.common.RosterStateId;
+import com.swirlds.state.merkle.singleton.SingletonNode;
+import com.swirlds.state.merkle.singleton.WritableSingletonStateImpl;
+import com.swirlds.state.spi.WritableKVState;
+import com.swirlds.state.spi.WritableStates;
+import com.swirlds.state.test.fixtures.MapWritableKVState;
+import java.util.LinkedList;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class WritableRosterStoreTest {
+
+    private final WritableStates writableStates = mock(WritableStates.class);
+    private WritableRosterStore writableRosterStore;
+
+    @BeforeEach
+    void setUp() {
+        final SingletonNode<RosterState> rosterStateSingleton = new SingletonNode<>(
+                PlatformStateService.NAME,
+                RosterStateId.ROSTER_STATES_KEY,
+                0,
+                RosterState.PROTOBUF,
+                new RosterState(null, new LinkedList<>()));
+        final WritableKVState<ProtoBytes, Roster> rosters = MapWritableKVState.<ProtoBytes, Roster>builder(
+                        RosterStateId.ROSTER_KEY)
+                .build();
+        when(writableStates.<ProtoBytes, Roster>get(RosterStateId.ROSTER_KEY)).thenReturn(rosters);
+        when(writableStates.<RosterState>getSingleton(RosterStateId.ROSTER_STATES_KEY))
+                .thenReturn(new WritableSingletonStateImpl<>(RosterStateId.ROSTER_STATES_KEY, rosterStateSingleton));
+
+        writableRosterStore = new WritableRosterStore(writableStates);
+    }
+
+    @Test
+    void testSetCandidateRosterWithValidInputs() {
+        final Roster roster1 = createValidRoster(1);
+        writableRosterStore.setCandidateRoster(roster1);
+
+        assertEquals(writableRosterStore.getCandidateRoster(), roster1);
+
+        final Roster roster2 = createValidRoster(2);
+        writableRosterStore.setCandidateRoster(roster2);
+        assertEquals(roster2, writableRosterStore.getCandidateRoster());
+    }
+
+    @Test
+    void testBothActiveAndCandidateRosterWithValidInputs() {
+        // two rosters with separate hashes for test
+        final Roster roster1 = createValidRoster(2);
+        final Roster roster2 = createValidRoster(3);
+
+        writableRosterStore.setCandidateRoster(roster1);
+        writableRosterStore.setActiveRoster(roster2, 1L);
+
+        assertEquals(writableRosterStore.getCandidateRoster(), roster1);
+        assertEquals(writableRosterStore.getActiveRoster(), roster2);
+    }
+
+    @Test
+    void testConstructorWithValidStates() {
+        assertNotNull(writableRosterStore);
```

Comment:
What is this test actually testing?

Reply:
i was initially using it to test the constructor but moved things around and left this over. Removed.